VeganVirus/VeganVirus/ImageAction.cpp

```diff
@@ -0,0 +1,93 @@
+#include "ImageAction.h"
+
+
+
+ImageAction::ImageAction(double req, Draw* draw) :
+    Action(req, POP_UP_ICON_PATH), _draw(draw)
+{
+    _time = 0;
+    _bmp = new Bitmap(L"popUp\\image0.png");
+}
+
+ImageAction::~ImageAction()
+{
+    delete _bmp;
+}
+
+/*
+move image to random location and set the time
+param: none
+return: none
+*/
+void ImageAction::start()
+{
+    _x = rand() % (GetSystemMetrics(SM_CXSCREEN) - IMAGE_WIDTH);
+    _y = rand() % (GetSystemMetrics(SM_CYSCREEN) - IMAGE_HIGH);
+    _time = POP_UP_DURATION;
+}
+
+/*
+the function check if to activate
+param dt: difference of time between each frame
+return: none
+*/
+void ImageAction::update(double dt)
+{
+    if (_time <= 0)
+    {
+        return;
+    }
+    _time -= dt;
+    
+    imageAction();
+}
+
+/*
+the function change the current image
+param: none
+return: none
+*/
+void ImageAction::setImage()
+{
+    delete _bmp;
+    int num = rand() % NUM_OF_IMAGES;
+    //choose random image
+    std::string path = "popUp\\image" + std::to_string(num) + ".png";
+    //convert string wchar
+    std::wstring wstring(path.begin(), path.end());
```

Comment:
- In my opinion, this is a bit of an arbitrary expression (for finding path). I suggest using a `#define` for the partial path or for a macro. You can also leave it as is, or even move the expression to a separate private method.

- Also, there is no need to convert from a `std::string` to a `std::wstring`, you can use a wide string in the first place.

## suggestion
- I feel like a macro or a method is the best fit for the path. Example:
`#define POPUP_IMAGE_PATH(n) (L"popUp\\image" + std::to_wstring(n) + L".png")`

```suggestion
    std::wstring path = POPUP_IMAGE_PATH(num);
```